README.md

```diff
@@ -2,100 +2,128 @@
+Just commit as usual. This module is triggered by a git commit-msg
+hook and automatically validates your messages as you commit
+them. Invalid messages will be rejected, with details on what's wrong
+and a copy of the input.
 
-By default, the commit message needs to match the [jQuery Commit Guidlines](http://contribute.jquery.org/commits-and-pull-requests/#commit-guidelines). See there for details, and below on how to change the default behaviour.
+The following ways to begin a commit message are special and always valid:
 
-In addition, a subject (the first line) starting with "fixup!" and "squash!" is considered valid. These are generated by `git commit --fixup` and `--squash`. A component in square brackets like "[Tmp]" or "[fix]" is also considered valid, to be manually squashed later. Commit messages for merges like "Merge branch [...]" or "Merge <commitish> into <commitish>" are considered valid as well.
+ 1. `0.0.1` or any other semantic version
+ 1. `WIP`, `Wip` or `wip` which means "work in progress"
+ 1. `Merge branch [...]` or `Merge <commitish> into <commitish>`
+ 1. `fixup!` or `squash!` which are generated by `git commit --fixup` and `--squash`
 
-## API
-
-```js
-var validate = require('commitplease/lib/validate');
-var errors = validate(commit.message);
-if (errors.length) {
-	postComment('This commit has ' + errors.length + ' problems!');
-}
-```
-
-`validate(message[, options])`, returns `Array`
+Common commit messages follow one of the style guides ([jQuery Commit
+Guidelines][1] by default)
 
-* `message` (`String`): The commit message to validate. Must use LF (`\n`) as line breaks.
-* `options` (`Object`, optional): Use this to override the default settings, see properties and defaults below
-* returns `Array`: Empty for valid messages, one or more items as `String` for each problem found
+## Setup
 
-Options and their defaults:
+You can configure commitplease from `package.json` of your
+project. Here are the options common for all style guidelines:
 
-```js
-component: true,
-components: [],
-markerPattern: '^(clos|fix|resolv)(e[sd]|ing)',
-actionPattern: '^([Cc]los|[Ff]ix|[Rr]esolv)(e[sd]|ing)\\s+[^\\s\\d]+(\\s|$)',
-ticketPattern: '^(Closes|Fixes) (.*#|gh-|[A-Z]{2,}-)[0-9]+',
-limits: {
-	subject: 72,
-	other: 80
+```json
+{
+  "commitplease": {
+    "limits": {
+      "limit0": "72",
+      "others": "80"
+    },
+    "nohook": false,
+    "markerPattern": "^(clos|fix|resolv)(e[sd]|ing)",
+    "actionPattern": "^([Cc]los|[Ff]ix|[Rr]esolv)(e[sd]|ing)\\s+[^\\s\\d]+(\\s|$)",
+    "ticketPattern": "^(Closes|Fixes) (.*#|gh-|[A-Z]{2,}-)[0-9]+",
+  }
 }
 ```
 
-* `component`: The default `true` requires a component, set to `false` to skip the check.
-* `components`: A list of valid components. When a component is found, it's compared to the ones specified in this array.
-* `limits`: Line length limits, for subject and other lines.
+ * `limits.limit0` and `limits.others` are the hard limits for the
```

Comment:
Let's rename this to `firstLine`, to be more explicit, but not style guide specific.